src/MentionsInput.js

```diff
@@ -358,6 +361,21 @@ class MentionsInput extends React.Component {
 
     let mentions = getMentions(newValue, markup, displayTransform, regex)
```

Comment:
You could do the same mentions extraction on the current `value` and then calculate the removed mentions as the difference between these two lists. This way you can get around keeping track of the mentions in an extra state field.

```
const previousMentions = getMentions(this.props.value, markup, displayTransform, regex)
const removed =  previousMentions.filter(...
```